src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Cmdlets/MachineLearningComputeCmdletBase.cs

```diff
@@ -0,0 +1,82 @@
+        public const string DescriptionParameterHelpMessage = "The operationalization cluster's description.";
+        public const string ClusterTypeParameterHelpMessage = "The operationalization cluster type.";
+        public const string OrchestratorTypeParameterHelpMessage = "The ACS cluster's orchestrator type.";
+        public const string ClientIdParameterHelpMessage = "The Kubernetes orchestrator service principal id.";
+        public const string SecretParameterHelpMessage = "The Kubernetes orchestrator service principal secret.";
+        public const string MasterCountParameterHelpMessage = "The number of master nodes in the ACS cluster.";
+        public const string AgentCountParameterHelpMessage = "The number of agent nodes in the ACS cluster.";
+        public const string AgentVmSizeParameterHelpMessage = "The VM size of the agent nodes in the ACS cluster.";
+        public const string StorageAccountParameterHelpMessage = "The URI to the storage account to use instead of creating one.";
+        public const string AzureContainerRegistryParameterHelpMessage = "The URI to the azure container registry to use instead of creating one.";
+        public const string ApplicationInsightsParameterHelpMessage = "The URI to the application insights to use instead of creating one.";
+        public const string ETagParameterHelpMessage = "The configuration ETag for updates.";
+        public const string GlobalServiceConfigurationAdditionalPropertiesHelpMessage = "Additional properties for the global service configuration.";
+        public const string SslStatusParameterHelpMessage = "SSL status. Possible values are 'Enabled' and 'Disabled'.";
+        public const string SslCertificateParameterHelpMessage = "The SSL certificate data in PEM format encoded as base64 string.";
+        public const string SslKeyParameterHelpMessage = "The SSL key data in PEM format encoded as base64 string.";
+        public const string SslCNameParameterHelpMessage = "The CName for the SSL certificate.";
+
+        public const string ClusterInputObjectAlias = "Cluster";
+
+        private IMachineLearningComputeManagementClient machineLearningComputeManagementClient;
+
+        public IMachineLearningComputeManagementClient MachineLearningComputeManagementClient
+        {
+            get
+            {
+                return machineLearningComputeManagementClient ??
+                    (machineLearningComputeManagementClient =
+                        AzureSession.Instance.ClientFactory.CreateArmClient<MachineLearningComputeManagementClient>(
+                            DefaultProfile.DefaultContext, Common.Authentication.Abstractions.AzureEnvironment.Endpoint.ResourceManager));
+            }
+        }
+
+        protected void HandleNestedExceptionMessages(CloudException e)
+        {
+            var exceptionDetails = e.Message + ".";
+
+            foreach (var err in e.Body.Details)
```

Comment:
@shutchings is the body ever null for an error? I want to make sure we avoid the scenario where `e.Body.Details` throws an error because `e.Body` is null.

Also, would we need to do any null-checking for `e.Body.Details`?

Reply:
Good points! I have to do this extra handling for the different types of errors that the SDK returns so that the error message to the user isn't just "Bad Request" or "Not Found". I've settled on trying to unwrap the errors as best as I can but I'll throw the original error if I have issues doing that. That way the user will always still get the original error, or a more descriptive one, and not an error due to the unwrapping.